Upstream: No, not upstreamable like this
Reason: We want the rustup binary in $CARGO_HOME/bin to be of the
same version as the one we distribute in this rustup package.
--- src/cli/self_update.rs.orig	2019-05-27 16:17:21.432715136 +0200
+++ src/cli/self_update.rs	2019-05-27 16:18:56.537718683 +0200
@@ -619,8 +619,7 @@
     if rustup_path.exists() {
         utils::remove_file("rustup-bin", &rustup_path)?;
     }
-    utils::copy_file(&this_exe_path, &rustup_path)?;
-    utils::make_executable(&rustup_path)?;
+    utils::symlink_file(&this_exe_path, &rustup_path)?;
     install_proxies()
 }
 
